i-VRESSE/bartender
# User roles

Refs #22 
Refs https://github.com/i-VRESSE/bartended-haddock3/issues/21

TODO
* [x] Add allowed_role to application config
* [x] Add roles to user table
* [x] Add grant/revoke routes
* [x] Add roles list route
* [x] Add roles to JWT token
* [x] Add /users route
* [x] Add roles to profile route
* [x] Check role on submit

To test
1. Add allowed_roles to config.yaml, see https://github.com/i-VRESSE/bartender/blob/user-roles/docs/configuration.md#applications
2. Register & login
3. Make yourself super user with `bartender super <email>`, so you can grant/revoke roles.
4. Grant role to user, 
    * Use GET /api/me to get own user id or GET /api/users to get user id of someone else
5. Verify with GET /api/users that role is granted
6. Revoke role from user
7. Verify with GET /api/users that role is revoked
8. Grant role which is not in GET /api/roles should give not found error.
9. Upload a job
    * Given users has no roles uploading a job should give forbidden error.
10. Grant a role to the user
11. Upload a job
    *. Given user with a role uploading a job should work.
12. An application without allowed_roles should be able to submitted to by any user with or without roles.
